### DISCUSSION TITLE
Setting file names for SBOMs

### DISCUSSION THREAD

**Discussion by MacOS** [oss-review-toolkit/ort#9585] [Q&A] [answered]
Hi all,

suppose you have the following call to the ``Reporter`` plugin, where I create a bunch of reports, including ``SPDX`` and ``CDX`` SBOMs.
```bash
ort/cli/build/install/ort/bin/ort \
        report \
                --report-option PlainTextTemplate=template.id=NOTICE_SUMMARY \
                --report-option SpdxDocument=spdxVersion=2.2 \
                --report-option SpdxDocument=outputFileFormats=JSON \
                --report-option CycloneDX=schema.version=1.5 \
                --report-option CycloneDX=output-file-formats=JSON,XML\
                --report-formats PlainTextTemplate,StaticHtml,WebApp,SpdxDocument,CycloneDx \
                --ort-file ort/evaluation-result.yml \
                --output-dir ort-output/
```

May I ask if, and if how, I can set the **file names** for the SBOMS? For example, I want them to have the following names.
- ``my-software-1.0.0.spdx.json``
- ``my-software-1.0.0.cdx.xml``
- ``my-software-1.0.0.cdx.json``

Thank you in advance!

**Comment by sschuberth** [comment, marked as answer]
There is no way to do that currently. One of the reasons is that being able to do so would require quite some additional CLI option logic to be able to set all names for all different formats of a specific reporter. Another reason is that it's trivial to rename files as a post-processing step.

**Reply by MacOS** [reply]
Thank you for the fast reply, @sschuberth.

I guess that means it is unlikely to be added any time soon?

**Reply by sschuberth** [reply]
Yes, there are no plans to add this feature. Which does not mean that we'd object to it if you come up with a sound concept how to implement it usage-wise, and contribute a PR afterwards.

**Reply by MacOS** [reply]
I understand - thank you for the fast answer again.

I may think of something and open an issue with a PR.
